### PR TITLE
Search/fetch courses

### PR DESCRIPTION
**Pull request by artiehumphreys**
New Trigram search logic for courses, including indexes over title and "<mnemonic> <number>". Deprecated regex search that can be costly computationally.

### CODE REVIEW

File: tcf_website/views/search.py [hunk 1]
```diff
@@ -131,47 +120,34 @@ def fetch_instructors(query):
...
-
-    # search query of form "<MNEMONIC><NUMBER>"
-    if number != "":
-        TITLE_WEIGHT = 0
-        MNEMONIC_WEIGHT = 1
-    # otherwise, "title" is entire query
-    else:
-        NUMBER_WEIGHT = 0
 
+    similarity_threshold = 0.3
```
Comment (barrett-ruth) [inline tcf_website/views/search.py:126]: magic

Comment (barrett-ruth) [reply tcf_website/views/search.py:126]: (explain)

Comment (barrett-ruth) [reply tcf_website/views/search.py:126]: also, how does this perform on searches for course titles? (i.e. not the mnemonics)

pls do just like 5 minutes of f*dging around with random searches n stuff before - once u say its fine we can merge

Comment (artiehumphreys) [reply tcf_website/views/search.py:126]: The 0.3 is for partial matches, in the case that a user searches a shorthand version of a class with a long name (ex: "data structures" when looking for "data structures and algorithms II"). Will do more testing now.

Comment (artiehumphreys) [reply tcf_website/views/search.py]: Changed similarity to 0.25 for a lower barrier for partial title searches. It seemed to work well with different searches.

**Review by barrett-ruth — APPROVED** [review]
nice

**Pull request merged**